adapters/repos/db/inverted/searcher_doc_pointers.go

```diff
@@ -75,7 +75,16 @@ func (fs *Searcher) docPointersInvertedNoFrequency(prop string, b *lsmkv.Bucket,
 			return false, errors.Wrap(err, "get hash")
 		}
 
-		hashes = append(hashes, currHash)
+		// currHash is only safe to access for the lifetime of the RowReader, once
+		// that has finished, a compaction could happen and remove the underlying
+		// memory that the slice points to. Since the hashes will be used to merge
+		// filters - which happens after the RowReader has completed - this can lead
+		// to segfault crashes. Now is the time to safely copy it, creating a new
+		// and immutable slice.
+		hashCopy := make([]byte, len(currHash))
```

Comment:
To avoid repetition and to encourage reuse I would define a small function

```
func copyBytes(bs []byte) []byte {
	new := make([]byte, len(bs))
	copy(new, bs)
	return new
}
```
We would end up with line below which is shorter and more expressive at same time
`hashes = append(hashes, copyBytes(currHash))`

Reply:
Good point, I'll add it.

Reply:
Done